### DISCUSSION TITLE
Client not authenticated when using Send-MailMessage

### DISCUSSION THREAD

**Discussion by Eds1989** [simonrob/email-oauth2-proxy#313] [Q&A] [answered]
I've got OAuth proxy setup and working fine for many of my internal services.

I am trying to add some email alerting to some PowerShell scripts, so am trying to use Send-MailMessage to submit a message to the OAuth proxy.
Whatever I try, I get this in the log:
`b'530 5.7.57 Client not authenticated to send mail. [LO4P123CA0409.GBRP123.PROD.OUTLOOK.COM 2024-12-16T18:23:13.136Z 08DD1CF27C1DC116]\r\n'`

Given the "client" is authenticated as it is working fine for other submitted messages, I'm confused why this is erroring?
Could anyone offer me any guidance please?

**Comment by simonrob** [comment]
Please post the proxy's log in [debug mode](https://github.com/simonrob/email-oauth2-proxy?tab=readme-ov-file#troubleshooting) so that I can see what authentication method is actually being attempted.

**Reply by Eds1989** [reply]
Sure thing. Here is the log output for when attempting to send a message using `Send-MailMessage`:
`2024-12-22 10:21:24,804: Setting debug mode: True`
`2024-12-22 10:22:08,806: New incoming connection to SMTP server at 0.0.0.0:1587 (unsecured) proxying smtp-mail.outlook.com:587 (STARTTLS)`
`2024-12-22 10:22:08,806: Accepting new connection from 10.0.0.51:51852 to SMTP server at 0.0.0.0:1587 (unsecured) proxying smtp-mail.outlook.com:587 (STARTTLS)`
`2024-12-22 10:22:08,837: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587) --> [ Client connected ]`
`2024-12-22 10:22:08,837: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'220 CWLP265CA0313.outlook.office365.com Microsoft ESMTP MAIL Service ready at Sun, 22 Dec 2024 10:22:08 +0000 [08DD222876954150]\r\n'`
`2024-12-22 10:22:08,837: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587) <-- b'220 CWLP265CA0313.outlook.office365.com Microsoft ESMTP MAIL Service ready at Sun, 22 Dec 2024 10:22:08 +0000 [08DD222876954150]\r\n'`
`2024-12-22 10:22:08,837: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587) --> b'EHLO Eds-MGMT\r\n'`
`2024-12-22 10:22:08,837: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     --> b'EHLO Eds-MGMT\r\n'`
`2024-12-22 10:22:08,853: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'250-CWLP265CA0313.outlook.office365.com Hello [193.237.202.172]\r\n'`
`2024-12-22 10:22:08,853: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'250-SIZE 157286400\r\n'`
`2024-12-22 10:22:08,853: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'250-PIPELINING\r\n'`
`2024-12-22 10:22:08,853: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'250-DSN\r\n'`
`2024-12-22 10:22:08,853: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'250-ENHANCEDSTATUSCODES\r\n'`
`2024-12-22 10:22:08,853: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'250-STARTTLS\r\n'`
`2024-12-22 10:22:08,853: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'250-8BITMIME\r\n'`
`2024-12-22 10:22:08,853: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'250-BINARYMIME\r\n'`
`2024-12-22 10:22:08,853: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'250-CHUNKING\r\n'`
`2024-12-22 10:22:08,853: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'250 SMTPUTF8\r\n'`
`2024-12-22 10:22:08,853: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     --> b'STARTTLS\r\n'`
`2024-12-22 10:22:08,868: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'220 2.0.0 SMTP server ready\r\n'`
`2024-12-22 10:22:08,868: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587) <-> [ Starting TLS handshake ]`
`2024-12-22 10:22:09,103: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587) [ Successfully negotiated SMTP server STARTTLS connection - re-sending greeting ]`
`2024-12-22 10:22:09,103: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     --> b'EHLO Eds-MGMT\r\n'`
`2024-12-22 10:22:09,134: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587) <-> [ TLSv1.3 handshake complete ]`
`2024-12-22 10:22:09,150: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'250-CWLP265CA0313.outlook.office365.com Hello [193.237.202.172]\r\n'`
`2024-12-22 10:22:09,150: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'250-SIZE 157286400\r\n'`
`2024-12-22 10:22:09,150: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'250-PIPELINING\r\n'`
`2024-12-22 10:22:09,150: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'250-DSN\r\n'`
`2024-12-22 10:22:09,150: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'250-ENHANCEDSTATUSCODES\r\n'`
`2024-12-22 10:22:09,150: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'250-AUTH LOGIN XOAUTH2\r\n'`
`2024-12-22 10:22:09,150: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'250-8BITMIME\r\n'`
`2024-12-22 10:22:09,150: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'250-BINARYMIME\r\n'`
`2024-12-22 10:22:09,150: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'250-CHUNKING\r\n'`
`2024-12-22 10:22:09,150: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'250 SMTPUTF8\r\n'`
`2024-12-22 10:22:09,150: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587) <-- b'250-CWLP265CA0313.outlook.office365.com Hello [193.237.202.172]\r\n250-SIZE 157286400\r\n250-PIPELINING\r\n250-DSN\r\n250-ENHANCEDSTATUSCODES\r\n250-AUTH PLAIN LOGIN\r\n250-8BITMIME\r\n250-BINARYMIME\r\n250-CHUNKING\r\n250 SMTPUTF8\r\n'`
`2024-12-22 10:22:09,165: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587) --> b'MAIL FROM:<eds89@outlook.com>\r\n'`
`2024-12-22 10:22:09,165: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     --> b'MAIL FROM:<eds89@outlook.com>\r\n'`
`2024-12-22 10:22:14,166: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587)     <-- b'530 5.7.57 Client not authenticated to send mail. [CWLP265CA0313.GBRP265.PROD.OUTLOOK.COM 2024-12-22T10:22:14.167Z 08DD222876954150]\r\n'`
`2024-12-22 10:22:14,166: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587) <-- b'530 5.7.57 Client not authenticated to send mail. [CWLP265CA0313.GBRP265.PROD.OUTLOOK.COM 2024-12-22T10:22:14.167Z 08DD222876954150]\r\n'`
`2024-12-22 10:22:14,166: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587) <-- [ Server disconnected ]`
`2024-12-22 10:22:14,166: SMTP (10.0.0.51:51852-{10.0.0.51:1587}-smtp-mail.outlook.com:587) --> [ Client disconnected ]`

**Reply by Eds1989** [reply, marked as answer]
Nevermind, me being dumb (half asleep while trying this).
Having had a read through the log entries for a successful send vs unsuccessul, I realised I wasn't providing credentials in the PowerShell command 🤦‍♂️
